plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/RemoveRedundantQualifierNameInspection.kt

```diff
@@ -153,6 +155,10 @@ private fun KtDotQualifiedExpression.applicableExpression(
     val newDescriptorFqName = newDescriptor.unwrapFakeOverrideIfNecessary().fqNameSafe
     if (originalDescriptorFqName != newDescriptorFqName) return null
 
+    if (newExpression is KtQualifiedExpression && newExpression.receiverExpression.declarationDescriptor(newContext) != receiverReference) {
```

Comment:
They should be checked by `compareDescriptors`. 
Please add a valid test on it
```kotlin
package m2.second

import m2.second.Bar.foo

data class Foo(val name: String)

object Bar {
  val foo = Foo("foo")
}

object Test {
  val boo = Bar.foo.copy(name = "foo2")
}
```

Reply:
`compareDescriptors` does not return the expected result because `compareDescriptorsText` returns true here.
https://github.com/JetBrains/intellij-community/blob/a29231587f989fb49f86af5246c9173413653088/plugins/kotlin/base/fe10/analysis/src/org/jetbrains/kotlin/idea/core/descriptorUtils.kt#L73

Reply:
Should we check `containingDeclaration`?

```kotlin
if (newExpression is KtQualifiedExpression &&
    !compareDescriptors(
        project,
        newExpression.receiverExpression.declarationDescriptor(newContext)?.containingDeclaration,
        receiverReference?.containingDeclaration
    )
) return null
```

Reply:
[150ebeb](https://github.com/JetBrains/intellij-community/pull/2120/commits/150ebeb4fea33ea84b5fbe8fc17c4ca730b0d0ed)